source/Calamari.Azure/AzureClient.cs

```diff
@@ -75,5 +76,26 @@ public static (ArmClientOptions, TokenCredentialOptions) GetArmClientOptions(thi
 
             return (armClientOptions, tokenCredentialOptions);
         }
+        public static async Task<AccessToken> GetAccessTokenAsync(this IAzureAccount azureAccount)
```

Comment:
I think this implementation came from the AzureAppService. There's a bit of logic missing around different AD domains, authorities, management endpoints and regions. Calamari.CloudAccounts includes a `GetAuthorizationToken` method in the `AzureOidcAccountExtensions` and `AzureServicePrincipalAccountExtensions` which handle this. 

I might be wrong and these tokens may be different but could we try and consolidate the logic here?

Reply:
Just as a side note, it's a tricky area in Calamari where we only test against Azure Public Cloud. There's also Government cloud, different management contexts and AD regions, it's a tricky spot with not much coverage.